packages/labs/task/src/task.ts

```diff
@@ -33,11 +33,12 @@ export const initialState = Symbol();
 
 export type TaskStatus = typeof TaskStatus[keyof typeof TaskStatus];
 
-export type StatusRenderer<R> = {
-  initial?: () => unknown;
-  pending?: () => unknown;
-  complete?: (value: R) => unknown;
-  error?: (error: unknown) => unknown;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type StatusRenderer<R, E, I = any, P = any, C = any, ER = any> = {
```

Comment:
Why do we need types for I, P, C, ER. Isn't the idea that we're inferring them?

Reply:
The `i, P, C, ER` is to hint to the TS compiler that these values should be infered.
The default value of `any` is so a type inference is not made at first, they could be `any` type.
When return values for `StatusRenderer` have a value of `unknown` then an `unknown` type is cast and anything that extends `StatusRenderer` will also have an `unknown` return type

Reply:
Heres a small demo of that behavior in action.

In the demo below, the function can't infer a type from values set as `unknown`. However in can infer types on values with a default of `any`
https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgKogNYgPYHcQBC2AHsgN4BQy1yAFhADYPYBcyArpjvgNxU208AEwCebTljwg+AXwoVQkWIhQB1RgwDS3QiXL9q9Jq2QAiI81MAaA3WHs254SOsU5FGJwRhg2EMmYAcyJiAB51JgARCBhQCCEQ5AhiSBAhAGc0LikQgD4ACgAjEjYIhmjYkHiQgEp9GmQoCDB2KH9i4gA6QVxRN3kEP3SwZA6ARjZhqFBA5ABeAOxgknyyCxNzDWxrO16xMx7RUxka+UVoeCRkAEEQEQAVEQAHapJQ+-nkODurZAAlT7fES5eoCLZse58BqHfZ-WRncAXFTIMraHJ6SjQ8EHLauaH2Rww1zuTwgby+fxBW4iELhDQVOIJPTJVIZG53R4vJnEAodUr0mKM2qg6hNFptUYkbpbeEUQYgYaS4gAJkmYGmIFmCypdxCq3WhNxvxhhOcxxqfHlio6AGY1RqtYtAtS9WtsZtjDsTQczScgA

Reply:
Here's an example of the slight difference between any and unknown when type inferencing.
Extending an interface with an `unknown` type actually casts a type of `unknown` and it needs to be reduced later somehow.

https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgKogNYgPYHcQBC2AHgDwAqyAvMgK6Y74A0yAStXQ3iAHzIDeAKGQjkACwgAbSdgBcyABQBKan3IBuYaLF4AJgE95y1W00BfQYNCRYiFAHUpkgNKNCJAVpETpcxSqo+ACIfGSCmL3E9Q38TEOjwwQtBGHoEMGBsEGQZAHMiMkdpABEIGFAIXQLkCGJIEF0AZzQufAKeBQAjEnkiyVLykEqCpXlWCDBaKBByfQAHCFI+gYqqkgBtEKdsIIBdPiFRZCgJqezu4gA6HVwDZXNLBCzGsGQLgEYOPIKFflC-YyBZBbXzhKK3GKA4I3AxBMxKSzWaDwJDIACCIH0swWazIlBocExLHYBMxB0i-yMATUmiOMMh1NMSUR4GRdmQAEkQDBoMMPIdtNsqXF-ok6dF5PEIYlkqkQOlMtk8hj9AUlk4VkNcTU6hAGs0Vdi+cQOhdehqyqsRmNTtMjeqSpatQVNqL9p4jidJtM3iRrtt7szBE8QC9fcQAEzyF5QUC5L7YXIqn5-IWxIEgsIsenCjP0uFKTQhsMXADM0bAsZA8ZoysxKcp6eh2zBOabwPz8KAA

Reply:
@justinfagnani maybe you might know of a more ergonomic way to pull types out of an interface supplied by a client?

Reply:
opted to use casting over any for type inference.